src/GHC/Util/Scope.hs

```diff
@@ -66,10 +67,10 @@ scopeMatch' (a, x) (b, y)
 -- scope that will refer to the same thing. If the resulting name is
 -- ambiguous, pick a plausible candidate.
 scopeMove' :: (Scope', Located RdrName) -> Scope' -> Located RdrName
-scopeMove' (a, x@(fromQual' -> Just name)) (Scope' b)
-  | null imps = head $ real ++ [x]
-  | all ideclQualified imps = noLoc $ mkRdrQual (unLoc $ head (mapMaybe ideclAs imps ++ map ideclName imps)) name
-  | otherwise = unqual' x
+scopeMove' (a, x@(fromQual' -> Just name)) (Scope' b) = case imps of
+  [] -> head $ real ++ [x]
+  (imp:_) | all ideclQualified imps -> noLoc $ mkRdrQual (unLoc . fromMaybe (ideclName imp) $ firstJust ideclAs imps) name
+          | otherwise -> unqual' x
```

Comment:
```suggestion
    _ -> unqual' x
```
Makes the completeness of the patterns more obvious

Reply:
Are you sure? I feel like this makes it less clear when the final case will be entered, and GHC not giving a warning is already evidence that it's complete.

Reply:
I'd have done it that way, but that's just my style - I'm not sufficiently fussed either way though. Let's go with how you wrote it.